### ISSUE TITLE
please create true comparisons with other whisper implementations

### ISSUE THREAD

**Issue by BBC-Esq** [sanchit-gandhi/whisper-jax#162]
Rather than re-typing everything, I'm simply providing a link to my issue request in the ```insanely-fast-whisper``` repository for you asking the same type of information be put in the readme:

[See Here](https://github.com/Vaibhavs10/insanely-fast-whisper/issues/82)

Plus, you guys both work on the same team so I'm hoping to work together to get some more accurate/explanatory numbers for people to rely on...

You claim that the Jax implementation is 70x faster...that portion of the readme hasn't been updated in awhile, and in the meantime other advancements have been made.  Also, it has never compared other options like faster-whisper, whisper.cpp, WhisperX, etc.  If you would be so kind, please include additional true comparisons for people, that way when they spend hours possibly revising code they can assess whether it's worth it from a cost-benefit perspective and they're clear on "batching" being the source of any speed increase...increased vram usage...or whatever.

Thanks, still love the work ya'll do!

**Comment by sanchit-gandhi** [comment]
Hey @BBC-Esq! Thanks for reaching out, I appreciate your interest in the Whisper-JAX project! Unfortunately this repo is more or less archived now, since we stopped working on it back in April. It was a fun project to see how fast we could make Whisper in JAX on TPU v4-8's, but the community is simply more interested in running on GPUs, which means we've switched to focussing on optimisations that can be applied uniformly, independent of hardware (e.g. Distil-Whisper: https://github.com/huggingface/distil-whisper). These are the results we got on an A100 with CUDA 11 and PyTorch 1.9 https://github.com/sanchit-gandhi/whisper-jax#benchmarks.

There are some scripts for reproducing the benchmarks [here](https://github.com/sanchit-gandhi/whisper-jax/tree/main/benchmarks). The `pmap` benchmarks should be run on at least a TPU v4-8, or v4-16 for the best performance on higher batch-sizes (where we are > 100x faster than Open AI Whisper).

**Comment by BBC-Esq** [comment]
Hey, thank you.  BTW, tell your colleague over at insanely-fast-whisper to change his readme and not dump on other peoples' work...

Moving on...Thank you sincerely for the technical discussion.  I'm excited that you're working on Distil-Whisper.  I still need to test that!

To make sure I understand you...Jax version is much faster on TPU but Hugging Face Transformers is much faster on GPUs?  I don't have a TPU so...

**Comment by BBC-Esq** [comment]
I forgot to ask, is it true that Distil-Whisper can't do any language besides English...and that's basically the tradeoff?

Looking forward to your work on distil-whisper.  Definitely in the hopper to try out.

**Comment by sanchit-gandhi** [comment]
We found Whisper JAX to be faster than Hugging Face Transformers' Whisper (same as `insanely-fast-whisper`) on our experiment. But those with different CUDA/PyTorch/Hardware versions had varying results, and sometimes PyTorch was faster. JAX is a real pain to set-up on CUDA, so I would encourage you to use a PyTorch implementation if you're working on GPU. With Flash Attention 2 support, and full torch compile support coming, I'm pretty sure Whisper in PyTorch will always beat Whisper in JAX in the next few weeks.

If you're using cloud computing, then swapping from GPUs to TPU v3's on GCP are quite reasonably priced, and you can run transcription super fast using there: https://cloud.google.com/tpu/pricing. TPU v4s are what we benchmarked to get the fastest results.

**Comment by sanchit-gandhi** [comment]
Yes that's right. Distil-Whisper is English-only since it's the language that has the most usage, but we still want to provide checkpoints that support more languages. Distilling Whisper on all the languages it supports in one go is hard - the decoder is very small, so it's difficult for it to have good knowledge of all languages at once. Instead, we're actively encouraging the community to distill language-specific Whisper checkpoints. We've released all the Distil-Whisper training code, and a comprehensive description of the steps required to perform distillation: https://github.com/huggingface/distil-whisper/tree/main/training. Feel free to ask if you have any questions about Distil-Whisper on the repo! I'd be more than happy to answer.

**Comment by BBC-Esq** [comment]
Let's try asking again...Did the dude at insanely-fast-whisper actually create any of the code of these underlying technologies?  Wink with your left eye if you don't want to answer because ya'll work together or wink with your right eye for no, he didn't actually create any new innnovation and it was other people...We'll just keep this between you and I.  lol.

**Comment by flexchar** [comment]
Based on https://github.com/Vaibhavs10/insanely-fast-whisper/issues/82 I'd suggest to have this closed.
